pkg/apiserver/logsearch/models.go

```diff
@@ -0,0 +1,141 @@
+	return json.Unmarshal([]byte(src.(string)), r)
+}
+
+func (r *SearchLogRequest) Value() (driver.Value, error) {
+	val, err := json.Marshal(r)
+	return string(val), err
+}
+
+type Component struct {
+	ServerType string `json:"server_type"`
+	IP         string `json:"ip"`
+	Port       string `json:"port"`
+	StatusPort string `json:"status_port"`
+}
+
+type TaskModel struct {
+	gorm.Model  `json:"-"`
+	TaskID      string            `json:"task_id" gorm:"unique_index"`
+	Component   *Component        `json:"component" gorm:"embedded"`
+	Request     *SearchLogRequest `json:"request" gorm:"type:text"`
+	State       TaskState         `json:"state"`
+	SavedPath   string            `json:"saved_path"`
+	TaskGroupID string            `json:"task_group_id"`
+	Error       string            `json:"error"`
+	CreateTime  int64             `json:"create_time"`
+	StartTime   int64             `json:"start_time"`
+	StopTime    int64             `json:"stop_time"`
+}
+
+type TaskGroupModel struct {
+	gorm.Model  `json:"-"`
+	TaskGroupID string
+	Request     *SearchLogRequest `json:"request" gorm:"type:text"`
+	state       TaskState
+}
+
+type PreviewModel struct {
+	gorm.Model `json:"-"`
+	TaskID     string                    `json:"task_id"`
+	Message    *diagnosticspb.LogMessage `json:"message" gorm:"embedded"`
+}
+
+type DBClient struct {
+	db *store.DB
+}
+
+var dbClient DBClient
+
+func (d *DBClient) initModel() {
+	d.db.AutoMigrate(&TaskModel{})
+	d.db.AutoMigrate(&TaskGroupModel{})
+	d.db.AutoMigrate(&PreviewModel{})
+}
+
+func (d *DBClient) createTask(task *TaskModel) error {
+	return d.db.Create(task).Error
+}
+
+func (d *DBClient) updateTask(task *TaskModel) error {
+	return d.db.Save(task).Error
+}
+
+func (d *DBClient) deleteTask(task *TaskModel) error {
+	return d.db.Delete(task).Error
+}
+
+func (d *DBClient) queryTaskByID(taskID string) (task TaskModel, err error) {
+	err = d.db.First(&task, "task_id = ?", taskID).Error
+	return
+}
+
+func (d *DBClient) queryTasks(taskGroupID string) (tasks []TaskModel, err error) {
+	err = d.db.Where("task_group_id = ?", taskGroupID).Find(&tasks).Error
+	return
+}
+
+func (d *DBClient) queryAllTasks() (tasks []TaskModel, err error) {
+	err = d.db.Find(&tasks).Error
+	return
+}
+
+func (d *DBClient) cleanAllUnfinishedTasks() error {
+	return d.db.Where("state != ?", StateFinished).Delete(&TaskModel{}).Error
+}
+
+func (d *DBClient) previewTask(taskID string) (previews []PreviewModel, err error) {
+	err = d.db.Where("task_id = ?", taskID).Find(&previews).Error
+	return
+}
+
+func (d *DBClient) newPreview(taskID string, msg *diagnosticspb.LogMessage) {
+	preview := PreviewModel{
+		TaskID:  taskID,
+		Message: msg,
+	}
+	d.db.Create(&preview)
+}
+
+func (d *DBClient) cleanPreview(taskID string) error {
+	return d.db.Delete(PreviewModel{}, "task_id = ?", taskID).Error
+}
+
+func (d *DBClient) cleanTask(taskID string) error {
+	return d.db.Delete(TaskModel{}, "task_id = ?", taskID).Error
+}
```

Comment:
I prefer to simply call inner methods directly, i.e. without introducing this abstraction, since the inner methods are already readable and simple enough.